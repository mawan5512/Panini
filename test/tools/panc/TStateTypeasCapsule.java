/*
 * This file is part of the Panini project at Iowa State University.
 *
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/.
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * For more details and the latest version of this code please see
 * http://paninij.org
 *
 * Contributor(s): Eric Lin
 */

/* @test
 * @summary Ensure that declaring states with capsule type gives a proper error message.
 * @compile/fail/ref=TStateTypeasCapsule.out -XDrawDiagnostics TStateTypeasCapsule.java
 */

capsule C11 {
    void consume() {
        System.out.println("C11");
    }
}

capsule C1 {
    C11 c11;
    void consume() {
        c11.consume();
        System.out.println("C1");
    }
}
