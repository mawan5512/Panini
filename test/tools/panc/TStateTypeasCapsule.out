TStateTypeasCapsule.java:32:9: compiler.err.states.with.capsule.type.error
1 error